### PR TITLE
Get rid of skopt

### PR DESCRIPTION
**Pull request by jasmainak**

### CODE REVIEW

File: autoreject/tests/test_bayesopt.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+def test_bayesopt():
+    """Test for bayesian optimization."""
+
+    x_star = 1.1
+
+    def func(x):
+        return (x - x_star) ** 2
+
+    grid = np.linspace(.0, 2.0, 30)
+    initial_x = grid[::5]
```
Comment (agramfort) [inline autoreject/tests/test_bayesopt.py]: initial_x = np.linspace(.0, 2.0, 5)

and no grid needed

File: autoreject/bayesopt.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+"""Bayesian optimization according to Brochu, Cora, and de Freitas' tutorial
+http://haikufactory.com/files/bayopt.pdf
+"""
+
+# Authors: Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+#          Mainak Jas <mainak.jas@telecom-paristech.fr>
```
Comment (agramfort) [inline autoreject/bayesopt.py]: add alex passos to the authors as I started from his code:

http://atpassos.me/post/44900091837/bayesian-optimization

Comment (jasmainak) [inline autoreject/bayesopt.py]: okay :) I was indeed wondering how you managed to implement it in 30 minutes ;)

**Comment by agramfort** [comment]
see that the diff with this code quite big though :)

**Comment by jasmainak** [comment]
The good thing is it stops Circle from timing out :) So, I can use continuous integration more easily to check PRs. And I also have the trick of cache directories, so it's much faster on Circle as well ...

**Comment by jasmainak** [comment]
@agramfort comments addressed. Good to go?

**Comment by agramfort** [comment]
LGTM

**Pull request merged**